Tests/SwiftLexicalLookupTest/NameLookupTests.swift

```diff
@@ -553,4 +579,132 @@ final class testNameLookup: XCTestCase {
       expectedResultTypes: .all(ClassDeclSyntax.self)
     )
   }
+
+  func testGuardOnFileScope() {
+    assertLexicalNameLookup(
+      source: """
+        let 1️⃣a = 0
+
+        class c {}
+
+        guard let 2️⃣a else { fatalError() }
+
+        3️⃣class a {}
+
+        let x = 4️⃣a
+        """,
+      references: [
+        "4️⃣": [
+          .fromFileScope(expectedNames: ["1️⃣"]),
+          .fromFileScope(expectedNames: ["3️⃣"]),
+          .fromScope(GuardStmtSyntax.self, expectedNames: ["2️⃣"]),
```

Comment:
Is the order of these meant to be meaningful? They're all effectively in the same scope, right?

Reply:
Yes, the only difference is that the first result represents the "member scope" within that file scope and the second one is from the "code block scope" part after the first non-declaration. 

As I mentioned in [this comment](https://github.com/swiftlang/swift-syntax/pull/2748#discussion_r1700084713), I'm not entirely sure now how should we handle this sort of sequential code that is e.g. used by playgrounds as it seems like all the names should be visible across the whole scope like in a member block.

Reply:
Thanks for clarifying.